apps/lockfile-explorer-web/src/App.tsx

```diff
@@ -1,37 +1,50 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect } from 'react';
+import styles from './appstyles.scss';
 import { readLockfile } from './parsing/readLockfile';
-import { LockfileEntry, LockfileEntryKind } from './parsing/LockfileEntry';
 import { LockfileViewer } from './containers/LockfileViewer';
 import { PackageJsonViewer } from './containers/PackageJsonViewer';
-import styles from './styles.scss';
+import { useAppDispatch } from './store/hooks';
+import { loadEntries } from './store/slices/entrySlice';
+import { LockfileEntryDetailsView } from './containers/LockfileEntryDetailsView';
+import { BookmarksSidebar } from './containers/BookmarksSidebar';
+import { SelectedEntryPreview } from './containers/SelectedEntryPreview';
+import { LogoPanel } from './containers/LogoPanel';
 
 /**
  * This React component renders the application page.
  */
 export const App = (): JSX.Element => {
-  const [selection, setSelection] = useState(LockfileEntryKind.Project);
-  const [projectEntries, setProjectEntries] = useState<LockfileEntry[]>([]);
-  const [packageEntries, setPackageEntries] = useState<LockfileEntry[]>([]);
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
     async function loadLockfile(): Promise<void> {
       const lockfile = await readLockfile();
-      setProjectEntries(lockfile.filter((l) => l.kind === LockfileEntryKind.Project));
-      setPackageEntries(lockfile.filter((l) => l.kind === LockfileEntryKind.Package));
+      dispatch(loadEntries(lockfile));
     }
     /* eslint @typescript-eslint/no-floating-promises: off */
     loadLockfile();
```

Comment:
This creates a floating promise. For example, if `readLockfile()` throws an exception, then the promise for `loadLockfile()` will reject and the runtime will report it as being uncaught.

Instead of supressing the ESLint warning, simply add a handler like:

```ts
    loadLockfile().catch((e) => {
      console.error(`Failed to read lockfile: ${e}`);
    });
```

(Ideally this error should get displayed somewhere visible to the end user, such as a dialog box in the web page, or the shell window where the CLI was invoked.  But that's something we could defer to a future PR if needed.)

Reply:
changed